policy-controller/k8s/index/src/pod.rs

```diff
@@ -72,48 +72,44 @@ pub(crate) fn port_names(spec: &Option<k8s::PodSpec>) -> HashMap<String, PortSet
 ///
 /// The result is a mapping for each probe port exposed by a container in the
 /// Pod and the paths for which probes are expected.
-pub(crate) fn get_http_probes(
-    spec: &k8s::PodSpec,
-    _port_names: &HashMap<String, PortSet>,
-) -> PortMap<HashSet<String>> {
-    let mut http_probes = PortMap::<HashSet<String>>::default();
-    for container in spec.containers.iter() {
-        let probes = (container.liveness_probe.iter())
-            .chain(container.readiness_probe.iter())
-            .chain(container.startup_probe.iter());
-        for probe in probes {
-            if let Some(ref http) = probe.http_get {
-                let path = http
-                    .path
-                    .as_ref()
-                    .expect("probe with httpGet should have a path field");
-                match http.port {
-                    k8s::IntOrString::Int(port) => {
-                        if let Ok(port) = u16::try_from(port).and_then(NonZeroU16::try_from) {
-                            let paths = http_probes.entry(port).or_default();
-                            paths.insert(path.clone());
-                        }
-                    }
-                    k8s::IntOrString::String(ref name) => {
-                        for port in container.ports.iter().flatten() {
-                            if let Some(ref n) = port.name {
-                                if n == name {
-                                    if let Ok(port) = u16::try_from(port.container_port)
-                                        .and_then(NonZeroU16::try_from)
-                                    {
-                                        let paths = http_probes.entry(port).or_default();
-                                        paths.insert(path.clone());
-                                    }
-                                    break;
-                                }
-                            }
-                        }
-                    }
+pub(crate) fn pod_http_probes(pod: &k8s::PodSpec) -> PortMap<HashSet<String>> {
+    let mut probes = PortMap::<HashSet<String>>::default();
+    for (port, path) in pod.containers.iter().flat_map(container_http_probe_paths) {
+        probes.entry(port).or_default().insert(path);
+    }
+    probes
+}
+
+fn container_http_probe_paths(container: &k8s::Container) -> Vec<(NonZeroU16, String)> {
```

Comment:
I think this benefits from being split into some smaller functions. This lets us use `?` short-circuiting.

Also, it seems simpler to collect all the port-paths and then join them into a set keyed on port.